Unite.Composer/Data/Datasets/Models/DatasetModel.cs

```diff
@@ -0,0 +1,35 @@
+using System.Text.Json.Serialization;
+
+namespace Unite.Composer.Data.Datasets.Models;
+
+public record DatasetModel
+{
+    private string _id;
+    private string _userId;
+    private string _domain;
+    private string _name;
+    private string _description;
+    private string _date;
+    private string _criteria;
+
+    [JsonPropertyName("key")]
+    public string Id { get => _id?.Trim(); set => _id = value; }
+
+    [JsonPropertyName("userid")]
```

Comment:
Multi-word property key "userid" should be "userId" here. Standard dotnet JSON parse should handle this.

Also, as id/key from the frontend you can send just "id" instead of "key", then dotnet model it will be mapped automatically and you won't even need to add [JsonPropertyName] attribute at all, dotnet JSON parser will handle it.

Frontend requires the "key", so it can be set somewhere here:
https://github.com/dkfz-unite/unite/blob/datasets/Unite.Web/Client/src/datasets/store/actions.js
In **load** method
state.datasets.forEach(dataset => {
      dataset.key = dataset.id;
      dataset.criteria = JSON.parse(dataset.criteria);
    });

For better consistency I'd change the "key" property to "id" on the frontend, updating all places where it's used, but this seems to require a lot of change, so i think it's not critical for now.

Reply:
Changing key property to id in the front end has more changes so that it can be implemented in the next features.